server/utilities.js

```diff
@@ -1,51 +1,22 @@
-var Promise = require('bluebird');
-var queries = require('./queries.js');
-var memoBuild = memoize(buildFunc);
+var memoize = module.exports.memoize = function(func) {
+  var cached = {};
 
-module.exports.evalAlg = function(userInput, dataType) {
-  var avgCutoff = 1000;
-  var avgIterations = 3;
-  var currentInputSize = 500;
-  var stepFactor = 1.2;
-  var result = [];
-  var runtime;
-
-  return new Promise(function(resolve, reject) {
-    return queries.getData(dataType)
-    .then(function(response) {
-      var data = response[0].array;
-
-      console.log('U18-data length is ' + data.length);
-
-      while (currentInputSize < 10000) {
-        if (currentInputSize <= avgCutoff) {
-          runtime = runTimeAverage(userInput, data.slice(0, currentInputSize), avgIterations);
-        } else {
-          runtime = getRunTime(userInput, data.slice(0, currentInputSize));
-        }
-        result.push(runtime);
-        currentInputSize *= stepFactor;
-      }
-
-      resolve(result);
-    });
-  });
-};
-
-module.exports.getJSONCoords = function(data) {
-  console.log('U36-getting json coordinates from eval data');
-
-  var coords = [];
-
-  for (var i = 0; i < data.length; i++) {
-    coords.push({x_axis: data[i][0], y_axis: data[i][1]});
-  }
+  return function() {
+    var args = Array.prototype.slice.call(arguments);
+    if (!cached[args]) {
+      cached[args] = func.apply(this, arguments);
+    }
 
-  return JSON.stringify(coords);
-};
+    return cached[args];
+  };
+}
```

Comment:
please add a semicolon to the end of this function variable declaration (line 12)